`dotnet publish` fails with `Microsoft.Azure.Functions.Worker.Sdk` version `1.17.0-preview3`

### Description

After the changes introduced in PR #2223, executing `dotnet publish` without the `local.settings.json` file now results in a failure. This is a critical issue for CI/CD scenarios that utilize GitHub Actions, as the absence of `local.settings.json` should not break the publishing process.

**Error message**

>error MSB3030: Could not copy the file "/home/runner/work/dotnet-isolated-deploy-test/dotnet-isolated-deploy-test/FunctionApp1/local.settings.json" because it was not found.

**Action history**

https://github.com/shibayan/dotnet-isolated-deploy-test/actions/runs/7728811909/job/21070518163

### Steps to reproduce

1. Update to `Microsoft.Azure.Functions.Worker.Sdk` version `1.17.0-preview3`

Repro code : https://github.com/shibayan/dotnet-isolated-deploy-test

--- a/sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.props
+++ b/sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.props
@@ -23,8 +23,8 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
   </ItemGroup>
 
   <ItemGroup>
-    <None Include="host.json" CopyToOutputDirectory="PreserveNewest" CopyToPublishDirectory="PreserveNewest" />
-    <None Include="local.settings.json" CopyToOutputDirectory="PreserveNewest" CopyToPublishDirectory="Never" />
+    <None Include="host.json" CopyToOutputDirectory="PreserveNewest" CopyToPublishDirectory="PreserveNewest" Condition="Exists('host.json')" />
+    <None Include="local.settings.json" CopyToOutputDirectory="PreserveNewest" CopyToPublishDirectory="Never" Condition="Exists('local.settings.json')" />
   </ItemGroup>
 
   <ItemGroup>

--- a/sdk/release_notes.md
+++ b/sdk/release_notes.md
@@ -7,3 +7,4 @@
 ### Microsoft.Azure.Functions.Worker.Sdk 1.17.0-preview4 (meta package)
 
 - Updating extension project generator to use `Microsoft.NET.Sdk.Functions` 4.3.0 (#2247)
+- Fix build when `local.settings.json` is not present (#2251)

